tests/components/sighthound/test_image_processing.py

```diff
@@ -91,3 +94,19 @@ def capture_person_event(event):
     state = hass.states.get(VALID_ENTITY_ID)
     assert state.state == "2"
     assert len(person_events) == 2
+
+
+async def test_save_image(hass, mock_image, mock_detections):
+    """Save a processed image."""
+    VALID_CONFIG[ip.DOMAIN].update({sh.CONF_SAVE_FILE_FOLDER: TEST_DIR})
+    await async_setup_component(hass, ip.DOMAIN, VALID_CONFIG)
+    assert hass.states.get(VALID_ENTITY_ID)
+
+    with patch("PIL.Image.open") as pil_img:
+        data = {ATTR_ENTITY_ID: VALID_ENTITY_ID}
+        await hass.services.async_call(ip.DOMAIN, ip.SERVICE_SCAN, service_data=data)
+        await hass.async_block_till_done()
+        state = hass.states.get(VALID_ENTITY_ID)
+        assert state.state == "2"
+        pil_img.save.assert_called()
```

Comment:
It's safer to not use the assert methods on the mock unless we spec the mock. A typo in the method would go unnoticed when running the test otherwise.
```suggestion
        assert pil_img.save.call_count == 1
```

Reply:
Hmm this is failing:
```
>           assert pil_img.save.call_count == 1
E           AssertionError: assert 0 == 1
E            +  where 0 = <MagicMock name='open().save' id='4622520728'>.call_count
E            +    where <MagicMock name='open().save' id='4622520728'> = <MagicMock name='open()' id='4622369288'>.save
```

Reply:
Oh yeah, we're calling img.convert too. So we need to account for that in the mocks.

Reply:
by additionally patching `homeassistant.components.sighthound.image_processing.Image.Image.convert`? I guess I should use a fixture

Reply:
No, since we're using the return value of Image.open which is a mock, we just need to adjust what object we're asserting.

```py
pil_img = pil_img_open.return_value
pil_img = pil_img.convert.return_value
```